source/server/admin/stats_params.cc

```diff
@@ -36,6 +36,29 @@ Http::Code StatsParams::parse(absl::string_view url, Buffer::Instance& response)
     return Http::Code::BadRequest;
   }
 
+  auto parse_type = [](absl::string_view str, StatsType& type) {
+    if (str == Labels::Gauges) {
+      type = StatsType::Gauges;
+    } else if (str == Labels::Counters) {
+      type = StatsType::Counters;
+    } else if (str == Labels::Histograms) {
+      type = StatsType::Histograms;
+    } else if (str == Labels::TextReadouts) {
+      type = StatsType::TextReadouts;
+    } else if (str == Labels::All) {
+      type = StatsType::All;
+    } else {
+      return false;
+    }
+    return true;
+  };
+
+  auto type_iter = query_.find("type");
+  if (type_iter != query_.end() && !parse_type(type_iter->second, type_)) {
+    response.add("invalid &type= param");
```

Comment:
maybe including the param value here will be helpful?

Reply:
In general I've learned to avoid including bogus input in an error response. It is an invitation to have an XSS.

Of course I can sanitize it but it's easier to just not include it.

Reply:
Sounds good.